File: salvar_numeros.c
#include <stdio.h>

#define TAM 15

void lerNumeros(float vetor[], int tamanho) {
    for (int i = 0; i < tamanho; i++) {
        printf("Digite o %dº número real: ", i + 1);
        scanf("%f", &vetor[i]);
    }
}

void salvarEmArquivo(float vetor[], int tamanho, const char *nomeArquivo) {
    FILE *arquivo = fopen(nomeArquivo, "w");
    if (arquivo == NULL) {
        printf("Erro ao abrir o arquivo.\n");
        return;
    }

    for (int i = 0; i < tamanho; i++) {
        fprintf(arquivo, "%.2f\n", vetor[i]);
    }

    fclose(arquivo);
    printf("Dados salvos em '%s'\n", nomeArquivo);
}

int main() {
    float vetor[TAM];
    lerNumeros(vetor, TAM);
    salvarEmArquivo(vetor, TAM, "dados.txt");
    return 0;
}

File: salvar_numeros.py
TAM = 15

def ler_numeros(tamanho):
    vetor = []
    for i in range(tamanho):
        while True:
            try:
                num = float(input(f"Digite o {i + 1}º número real: "))
                vetor.append(num)
                break
            except ValueError:
                print("Entrada inválida. Tente novamente.")
    return vetor

def salvar_em_arquivo(vetor, nome_arquivo):
    try:
        with open(nome_arquivo, "w") as f:
            for numero in vetor:
                f.write(f"{numero:.2f}\n")
        print(f"Dados salvos em '{nome_arquivo}'")
    except IOError:
        print("Erro ao salvar o arquivo.")

def main():
    vetor = ler_numeros(TAM)
    salvar_em_arquivo(vetor, "dados.txt")

if __name__ == "__main__":
    main()